src/lib/eslint/EslintProcessHelper.ts

```diff
@@ -0,0 +1,77 @@
+import { CLIEngine } from 'eslint';
+import * as path from 'path';
+import { CUSTOM_CONFIG } from '../../Constants';
+import { RuleResult, RuleViolation, ESMessage, ESRule, ESReport } from '../../types';
+
+
+export class StaticDependencies {
+	/* eslint-disable @typescript-eslint/no-explicit-any */
+	createCLIEngine(config: Record<string, any>): CLIEngine {
+		return new CLIEngine(config);
+	}
+
+	resolveTargetPath(target: string): string {
+		return path.resolve(target);
+	}
+
+	getCurrentWorkingDirectory(): string {
+		return process.cwd();
+	}
+}
+
+export class EslintProcessHelper {
```

Comment:
I'm not in love with this name. Is there a more descriptive name we could use instead? Alternatively, would it have been possible to structure the code so `CustomEslintEngine` extends `BaseEslintEngine`, and therefore has all of the same methods without the need for a helper?

Reply:
I initially started with making changes to `BaseEslintEngine` to accommodate the custom config and the outcome was confusing and buggy. Having a separate engine for handling the custom config seemed a lot simpler and safe from regressions.

To avoid code duplication, I followed the common design principle of favoring [Composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance).

The name of this module could be nicer for sure - not a fan of `Helper` either. Let me know if you have any ideas.